core/src/main/scala/org/apache/spark/sql/aqp/SnappyContextFunctions.scala

```diff
@@ -33,7 +34,18 @@ class SnappyContextFunctions {
 
   def postRelationCreation(relation: Option[BaseRelation], session: SnappySession): Unit = {}
 
-  def registerAQPErrorFunctions(session: SnappySession) {}
+  def registerSnappyFunctions(session: SnappySession): Unit = {
+    val registry = session.sessionState.functionRegistry
+
+    registry.registerFunction("CURRENT_USER",
+      e => {
+        if (! e.isEmpty) {
+          throw new AnalysisException("Argument(s)  passed for zero arg function " +
+              s"CURRENT_USER")
+        }
+        CurrentUser()
+      })
+  }
```

Comment:
I think this was specifically for AQP functions. There is another list in SnappyDataFunctions.registerSnappyFunctions. Also adding use the overload with "info" field then it will appear in DESCRIBE FUNCTION.

Reply:
Oh I did not know that . So I renamed the function name used for registering aqp functions to indicate common usage. I also incidentally renamed it as registerSnappyFunctions. Will move it to functions.scala after the merge. Adding the info.